File: Problem2.m
%CHEME 5440
%Problem 2
%part e
%Parameters
cao = 1;
cro = 10;
xlimca = [0 200 10];
ylimcr = [0 100 10];
xinit = [cao cro];

ror = 1;
rr = 100;
roa = 100;
ra = 5000;
da = 30;

f = @(t,c) [-da*c(1) + (roa+ra*c(1)^2)/(1+c(1)^2+c(2)^2); ...
    -c(2) + (ror+rr*c(1)^2)/(1+c(1)^2)];
[t,cf] = ode45(f,[0 50],[1 10]);

figure
plot(t,cf(:,1))
hold on
plot(t,cf(:,2))
title("time versus concentration")
legend("Species A","Species B")
xlabel("time")
ylabel("concentration")
ylim([0 200])

File: Problem2.jl
# CHEME 5440/7770
# Problem 2
include("PhasePortraitV2.jl")

function odefun(ca, cr)
    # Parameters
    ror = 1.0
    rr = 100.0
    roa = 100.0
    ra = 5000.0
    da = 30.0

    u = @. -da*ca + (roa+ra*ca^2)/(1+ca^2+cr^2)
    v = @. -cr + (ror+rr*ca^2)/(1+ca^2)

    return(u,v)
end


tspan = (0.0, 50.0)

xlimca = (0, 200, 50)
ylimcr = (0, 100, 50)
xo = ([1.0, 10.0],)

phaseplot(odefun,xlimca,ylimcr,clines=true,xinit=xo,t=tspan,norm=true,scale=0.5)
